src/compiler/binder.ts

```diff
@@ -3083,7 +3083,7 @@ namespace ts {
         }
 
         function bindPropertyAssignment(name: BindableStaticNameExpression, propertyAccess: BindableStaticAccessExpression, isPrototypeProperty: boolean, containerIsClass: boolean) {
-            let namespaceSymbol = lookupSymbolForPropertyAccess(name);
+            let namespaceSymbol = lookupSymbolForPropertyAccess(name, container) || lookupSymbolForPropertyAccess(name, blockScopeContainer);

```

Comment:
silly question: what happens if you *just* use blockScopeContainer? I assume that some other tests break.

Reply:
I was pairing with Orta on this and I did try this, and stuff did fail. It was a little surprising to me, but I didn’t have time to dig into why that was the case.

Reply:
`blockScopeContaininer` only contains the block scoped stuff. function-scoped stuff (like function declarations and `var`s), will only be in the function container (assuming they're different).